src/main/groovy/com/deploygate/gradle/plugins/DeployGatePlugin.groovy

```diff
@@ -37,19 +43,37 @@ class DeployGatePlugin implements Plugin<Project> {
     void apply(Project project) {
         DeprecationLogger.reset()
 
-        setupExtension(project)
+        def credentialStore = new CliCredentialStore()
+        def extension = setupExtension(project, credentialStore)
+
         GradleCompat.init(project)
         AndroidGradlePlugin.init(project)
         initProcessor(project)
 
+        project.tasks.register(Constants.LOGIN_TASK_NAME, LoginTask) {
+            it.description = "Check the configured credentials and launch the authentication flow if they are not enough."
+
+            it.group = Constants.TASK_GROUP_NAME
+            it.deployGateExtension = extension
```

Comment:
We can set properties like this. This means that configuring the required values (inputs) at the task registration time.